### PR TITLE
AAP-88660 XLAB Implement gamification UI components against mock data

### PR DESCRIPTION
**Pull request by kcagran**
## Summary
AAP-88660  Implement gamification UI components against mock data
<!-- What changed and why? -->
This pull request implements the **Leaderboards** tab — an automation gamification view — on the AWX Automation Dashboard. The Automation Dashboard becomes a tabbed page, and the layout and filter-handling refactors needed to support a second tab are included. The Leaderboards tab runs against mock data until an analytics endpoint exists.         
                                                                                                                                                                                                                                                                                                                                                               
  **New Leaderboards tab (gamification UI):**                                                                                                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                                                                                                               
  - Added `AutomationLeaderboards` and the `components/leaderboards/` component set: at-a-glance KPIs (`AutomationAtAGlance`, `AtAGlanceKpiMetric`), automation dimensions with a per-dimension Top-10 (`AutomationDimensions`), Top-10 organizations (`HighlightsLeaderboardPanel`, `LeaderboardRankCell`), the 30-day automation-streak calendar             
  (`StreakDayStrip`), milestone badges (`MilestoneBadgesCard`), the last-sync line (`HighlightsSyncTimestamp`), and shared text/heading primitives.                                                                                                                                                                                                            
  - All data comes from one hook, `useAutomationLeaderboardsView`, which owns the full data contract and currently returns a `MOCK_LEADERBOARDS` constant. When the endpoint lands, only the hook body changes.                                                                                                                                                
  - Loading (`isLoading`) and empty (`lastSyncedAt === null`, via framework `EmptyStateNoData`) states are handled.                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                                                                                               
  **Tabbed dashboard shell:**                                                                                                                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                                                                                                               
  - New `AutomationDashboardMainPage` renders `PageRoutedTabs` with **Dashboard** (the existing dashboard, unchanged) and **Leaderboards** tabs, plus the shared `PageHeader` and the collection-status loading gate.                                                                                                                                          
  - Routing: `AwxRoute.AutomationDashboard` enum value is unchanged (now the `dashboard` child route); new `AutomationDashboardMainPage` / `AutomationLeaderboards` routes; `<Navigate to="dashboard" replace>` for the index. Platform navigation references the new parent id instead of a hard-coded string.                                                
                                                                                                                                                                                                                                                                                                                                                               
  **Grid layout refactor:**                                                                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                                                               
  - New `useDashboardGridColumns` measures a zero-height probe in the persistent page shell once and exposes the column count via `PageDashboardContext`. New `DashboardLayout` / `DashboardGridRow` replace the per-page `ref` + `useResizeObserver` + context boilerplate in both tabs, so switching tabs no longer re-measures or flashes the grid.         
                                                                                                                                                                                                                                                                                                                                                               
  **Filter and report session persistence:**                                                                                                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                                                               
  - New `utils/persistedFilterState.ts` persists the Dashboard toolbar filter state and the selected "Select report" filter set to `sessionStorage`, keyed per active user so a second login on the same tab cannot inherit the previous user's filters. A URL deep link still wins on first load. Includes type guards for the untrusted JSON, reused by      
  `DashboardToolbar`.                                                                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                                                               
  **Bug fixes:**                                                                                                                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                                                               
  - `useAutomationDashboardView` — the default filter embedded start/end dates resolved at module load, so after a UTC day rollover + reload the default view showed as an active custom filter. The `last_7_days` preset now carries no dates (they were never sent for non-custom presets); both `isDefaultFilterState` implementations now agree.           
  - `useFilterSetView` — `queryOptions` only appended new ids, so a filter set seeded from `sessionStorage` that was renamed or re-filtered server-side never refreshed. It now reconciles cached entries and the active selection against the server response.                                                                                                
                                                                                                                                                                                                                                                                                                                                                               
  **Cleanup:**                                                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                                                               
  - Removed the dead "Top 5 projects" / "Top 5 users" `DashboardTableCard` (component, type, tests, export).                                                                                                                                                                                                                                                   
  - Grid-measurement and loading/empty-state logic moved out of `AutomationDashboard` / its test into the shell.                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                                                               
  **Styling:**                                                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                                                               
  - New `AutomationDashboard.css` for the streak heatmap, achievement badges, grid layouts and leaderboard crowns, using PatternFly design tokens.                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                                                                                               
  **Tests:**                                                                                                                                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                                                               
  - 17 new test files; the full `automation-dashboard` suite (478) and platform `useAutomationAnalytics` (24) pass. `tsc` (awx + platform), `eslint`, `prettier` clean. 
## Type of Change

- [ ] Bug fix
- [x] Enhancement
- [x] Tests
- [x] Documentation
- [ ] Other (please specify)

## Risk Analysis - REQUIRED

<!-- SELECT ONE. This is required — the PR check will fail if none is selected. -->

- [ ] **High** — Broad platform impact (e.g., SWR config, shared framework components, authentication, routing, API wrappers, build/bundler config). Changes affect multiple workspaces or could cause widespread regressions.
- [x] **Medium** — Scoped but cross-cutting (e.g., shared utility functions, changes to multiple pages within one workspace, component library updates, test infrastructure). Limited blast radius but touches common code paths.
- [ ] **Low** — Narrowly scoped (e.g., single page fix, styling tweak, documentation, test-only changes, copy/string updates). Minimal risk of unintended side effects.

## Dependencies

<!-- List any dependent PRs, required backend changes, or manual setup steps. Remove this section if none. -->

## Testing

### Ephemeral E2E Tests

Once PR is ready and preliminary checks pass, trigger tests by posting a comment `/run-playwright` on this PR.

> Tests run against a fresh AAP instance (version based on branch).

### External Server E2E Runs

<!-- If applicable, attach run(s) from an external server using the GitHub Actions below. Mention the deployment type of the environment used.
- [Run Playwright with Currents](../actions/workflows/run-playwright-currents.yml)
-->

### Manual Testing

<!-- Steps to verify this change, if not obvious from the Jira issue. -->

### Screenshots

<!-- If applicable. Any visual/UI changes MUST include before and after screenshots. -->
<img width="2007" height="1888" alt="image" src="https://github.com/user-attachments/assets/22e92507-0b11-4389-a499-a40d03658a07" />

<img width="1737" height="528" alt="image" src="https://github.com/user-attachments/assets/a9607da0-ebe3-49e9-a9bb-e437a294a028" />

### CODE REVIEW

**Review by tiyiprh — COMMENTED** [review]
UX review — looks good for mock-data phase!

Reviewed against the ANSTRAT-1976 Leaderboards prototype. Overall this matches the intended experience well: tabbed Dashboard/Leaderboards shell, section order (KPIs → streaks → dimensions → org top-10 + milestone badges), mock hook pattern, empty/loading states, and framework usage (PageDashboardCard, PageTable, measured grid) all look right.

Two small fixes in this PR:
1. Card title should be "Automation at a glance" (sentence case) — currently Automation at a Glance in AutomationAtAGlance.tsx

2. The success fill looks correct (--pf-t--global--color--status--success--default). The no activity state does not.

Issue: Using a hardcoded light gray (e.g. #e0e0e0) or only --pf-t--global--background--color--tertiary--default won’t work across light and dark. Hardcoded hex is light-mode-only; tertiary is a surface token and can blend into the card instead of reading as “empty day.”

Pattern to follow: Same approach as other neutral fills in this feature (e.g. silver tier badges in postGa.css):
Success: var(--pf-t--global--color--status--success--default)
No activity (light): var(--pf-t--global--color--nonstatus--gray--400)
No activity (dark): var(--pf-t--global--color--nonstatus--gray--300) under .pf-v6-theme-dark
Diagonal stripe: color-mix with var(--pf-t--global--border--color--default) (not fixed rgba(0,0,0,…))
Define shared CSS vars on .post-ga-highlights-tab (e.g. --post-ga-streak-success-color, --post-ga-streak-empty-color) and use them for both the heat cells and the legend swatches so they stay in sync.
Where: postGa.css (.streak-heat-cell--success / --empty), AutomationStreakCard.tsx (legend colors)


There have been some updates to the prototype but still pending confirmation from PM + team so that will likely be a separate issue but just wanted to let you know for awareness.

_Cursor assisted_

**Review by tiyiprh — APPROVED** [review]
LGTM from UX perspective - changes were made. TY!

File: frontend/awx/main/useAwxNavigation.tsx [hunk 1]
```diff
@@ -108,10 +110,26 @@ export function useAwxNavigation() {
...
+              element: <AutomationDashboard />,
+            },
+            {
+              id: AwxRoute.AutomationLeaderboards,
+              path: 'leaderboards',
+              element: <AutomationLeaderboards />,
+            },
+            {
+              path: '',
+              element: <Navigate to="dashboard" replace />,
```
Comment (lgalis) [inline frontend/awx/main/useAwxNavigation.tsx]: The dashboard URL moved from `/analytics/automation-dashboard` to `/analytics/automation-dashboard/dashboard`. This index redirect uses a pathname-only `to`, so React Router drops `search`.

Existing bookmarks such as `/analytics/automation-dashboard?period=last_30_days` will land on `/dashboard` with no query, then the view seeds last-7-days (or sessionStorage). The PR notes that a URL deep link still wins on first load — that is only true for the new child path, not the previous public URL.

Preserve the query string, e.g. `to={{ pathname: 'dashboard', search: location.search }}`.

File: frontend/awx/analytics/automation-dashboard/views/useFilterSetView.tsx [hunk 2]
```diff
@@ -1,24 +1,79 @@
...
+  useEffect(() => {
+    if (userId === undefined || seededUserId === userId) return;
+    setSeededUserId(userId);
+    const persisted = readPersistedFilterSet(userId);
+    setValue(persisted ? String(persisted.id) : undefined);
+    setSelectedFilterSet(persisted);
+    setVersion((v) => v + 1);
+    setFilterSets((prev) => {
+      if (!persisted) return prev;
+      return prev.some((fs) => fs.id === persisted.id) ? prev : [persisted, ...prev];
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/views/useFilterSetView.tsx]: `value` / `selectedFilterSet` are re-seeded when the active user changes, but `filterSets` is not reset. If the new user has nothing persisted, `return prev` keeps the previous user's saved reports. If they do, the new set is prepended onto the old cache.

Logout/login in the same tab is the case this persistence was added for — User B can briefly (or until the next fetch) see User A's report names and `filters` JSON in the dropdown cache. The user-switch test does not assert `filterSets`.

Replace the cache on seed (`persisted ? [persisted] : []`) and cover it in the existing user-switch test.

File: frontend/awx/analytics/automation-dashboard/components/leaderboards/StreakDayStrip.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+              content={streakDayTooltipContent(day, success, runs, t)}
+              position="top"
+            >
+              <div
+                className={
+                  success
+                    ? 'streak-heat-cell streak-heat-cell--success'
+                    : 'streak-heat-cell streak-heat-cell--empty'
+                }
+                aria-label={streakDayTooltipAriaLabel(day, success, runs, t)}
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/components/leaderboards/StreakDayStrip.tsx:120]: These heatmap cells are empty `<div>`s with `aria-label` and no `role`. Testing Library will still find them via `getByLabelText`, but many screen readers do not expose `aria-label` on a generic div. They are also not in the tab order, so the run-count tooltip is mouse-only.

Use a focusable control (or `role="img"` plus `tabIndex={0}`) so the accessible name is exposed and keyboard users can reach the tooltip.

File: frontend/awx/analytics/automation-dashboard/components/leaderboards/MilestoneBadgesCard.tsx [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+            ? 'achievement-badge achievement-badge--earned achievement-badge--milestone'
+            : 'achievement-badge achievement-badge--locked achievement-badge--milestone'
+        }
+        aria-label={badge.label}
+      >
+        <span className="achievement-badge__stars" aria-hidden />
+        <span className="achievement-badge__icon">{badge.icon}</span>
+        <span className="achievement-badge__label">{badge.label}</span>
+      </div>
+    </Tooltip>
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/components/leaderboards/MilestoneBadgesCard.tsx:161]: The tooltip wrapper is a non-focusable `div`, so keyboard users cannot read the badge rule. `aria-label` is only `badge.label`, so earned vs locked is not announced.

Make the trigger focusable and include earned/locked (and ideally the rule) in the accessible name.

File: frontend/awx/analytics/automation-dashboard/AutomationLeaderboards.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      <DashboardGridRow>
+        <HighlightsLeaderboardPanel width={bottomCardsWidth} />
+        <MilestoneBadgesCard width={bottomCardsWidth} />
+      </DashboardGridRow>
+    </>
+  );
+}
+
+export function AutomationLeaderboards() {
+  const { lastSyncedAt, isLoading } = useAutomationLeaderboardsView();
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/AutomationLeaderboards.tsx]: The view type includes `error`, but this only branches on `isLoading` and `lastSyncedAt === null`. When the hook is swapped to SWR, a failed fetch will not surface anything useful.

Please add an error empty state now (even if the mock always returns `undefined`), and a test that the loaded layout mounts (sync line + cards). `AutomationLeaderboards.test.tsx` currently covers loading and never-synced only.

File: frontend/awx/analytics/automation-dashboard/components/leaderboards/AutomationDimensions.tsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        </FlexItem>
+        <FlexItem style={{ textAlign: 'right' }}>
+          <MetricValue>
+            {rank <= 3 ? (
+              <CrownIcon
+                className={LEADERBOARD_RANK_CROWN_CLASS[rank as 1 | 2 | 3]}
+                style={{ marginRight: 6, verticalAlign: '-0.05em' }}
+              />
+            ) : null}
+            {score.toLocaleString()}
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/components/leaderboards/AutomationDimensions.tsx]: At-a-glance KPIs and the org leaderboard use `toLocaleString(DEFAULT_NUMBER_LOCALE)` (`en-US`). These dimension scores use the browser locale, so the same page can show `1,234` and `1234`.

Please use `DEFAULT_NUMBER_LOCALE` here and on `row.value.toLocaleString()` below (line 127).

**Review by lgalis — COMMENTED** [review]
Thanks for the thorough mock-data Leaderboards pass — tabbed shell, session-scoped filter persistence, and the grid measurement probe all look like the right approach. UX already signed off; notes below are from a code review.

## Requested in this PR

1. **Index redirect drops query-string deep links** — `/analytics/automation-dashboard?period=…` will lose filters when redirected to `/dashboard`.
2. **Filter-set cache leaks across users** — `filterSets` is not reset on user switch, which undercuts the user-keyed sessionStorage work.
3. **Keyboard / AT on streak cells and milestone badges** — `aria-label` on non-focusable `div`s; earned vs locked not announced.
4. **Leaderboards error path + loaded-layout test** — `error` is on the view type but unused; tests never mount the happy path.
5. **Number formatting** — dimension scores should use `DEFAULT_NUMBER_LOCALE` like the rest of this feature.

## Follow-ups (not blocking this PR)

- Child cards each call `useAutomationLeaderboardsView`. Fine while the hook is a constant; keep **one SWR key** when the API lands so this does not fan out five fetches.
- `AutomationDashboard.css` still carries unused prototype rules (96px badges, star row always hidden on milestones). Safe to trim.
- `useAutomationLeaderboardsView` tests assert fixture values (`jobsRun === 1234`, `'Jamie Ortiz'`). Those will all churn when the endpoint is wired — prefer contract tests (shape, empty, error, loading) and keep the mock in a fixture file.

**Review by lgalis — APPROVED** [review]
Medium - can be added in a follow-up PR
1. Loaded Leaderboards layout still untested

AutomationLeaderboards.test.tsx covers loading, empty, and error paths, but not the happy path (sync line + card sections). You asked for this; child components are tested individually, but the page-level wiring is not.

Suggestion: Add one test with a minimal mocked view (non-zero lastSyncedAt, no error) asserting sync timestamp + key section headings (Automation at a glance, dimensions, org leaderboard, achievements).

**Comment by SherinV** [comment]
> Medium - can be added in a follow-up PR
> 
> 1. Loaded Leaderboards layout still untested
> 
> AutomationLeaderboards.test.tsx covers loading, empty, and error paths, but not the happy path (sync line + card sections). You asked for this; child components are tested individually, but the page-level wiring is not.
> 
> Suggestion: Add one test with a minimal mocked view (non-zero lastSyncedAt, no error) asserting sync timestamp + key section headings (Automation at a glance, dimensions, org leaderboard, achievements).

@appuk Please plan to add the test @lgalis recommended. Thanks.

**Comment by kcagran** [comment]
> > Medium - can be added in a follow-up PR
> > 
> > 1. Loaded Leaderboards layout still untested
> > 
> > AutomationLeaderboards.test.tsx covers loading, empty, and error paths, but not the happy path (sync line + card sections). You asked for this; child components are tested individually, but the page-level wiring is not.
> > Suggestion: Add one test with a minimal mocked view (non-zero lastSyncedAt, no error) asserting sync timestamp + key section headings (Automation at a glance, dimensions, org leaderboard, achievements).
> 
> @appuk Please plan to add the test @lgalis recommended. Thanks.

Will be implemented in the next PR (design change). PR will be ready for review tomorrow.
